src/lightning/pytorch/trainer/call.py

```diff
@@ -73,6 +73,10 @@ def _call_setup_hook(trainer: "pl.Trainer") -> None:
     assert trainer.state.fn is not None
     fn = trainer.state.fn
 
+    # Trigger lazy creation of experiment in loggers so loggers have their metadata available
+    for logger in trainer.loggers:
+        _ = logger.experiment
```

Comment:
Why not initialize it when `loger.foo` is accessed for the first time (if it's not initialized already)?

Reply:
Because before `logger.experiment.foo` is accessed for the first time, `logger.version` and `logger.name` will be None. The model checkpoint for example would create a checkpoint dir with None as the name/version.